rewrite-java/src/main/java/org/openrewrite/java/internal/template/BlockStatementTemplateGenerator.java

```diff
@@ -803,6 +806,13 @@ public J visitMethodInvocation(J.MethodInvocation method, Integer integer) {
                     //noinspection ConstantConditions
                     return mi.getSelect();
                 }
+                if (method.getTypeParameters() != null) {
+                    // For method chains return `select` if `STOP_COMMENT` is found before `typeParameters`
+                    if (stopCommentExists(mi.getPadding().getTypeParameters().getBefore().getComments())) {
+                        //noinspection ConstantConditions
+                        return mi.getSelect();
+                    }
+                }
```

Comment:
This is what fixed the unit test added here, for the minimal possible impact change we could do here.

A larger change could reduce remove the `__TEMPLATE_STOP__` added twice, but it's unclear what unintended side effects that might have downstream, so I went for this safer option.